### PR TITLE
Use localstorage instead of cookies for password hash

### LINKED ISSUE #265: Store passwordHash in localstorage and only send it when needed to backend
This way we don't have to force a reload after registration. 

It's only used in three places:
- On initial connection
- When voting on anonymous issues
- When registering

Fixing the initial connection case is a bit trickier.

### PR DESCRIPTION
**Pull request by duvholt**
Fixes #265

Implemented using a separate websocket event and only sends passwordhash to backend when needed.
Client reply is implemented using a saga that simply waits for the 'request action' and responds with a 'send action'.

### CODE REVIEW

File: server/utils/socketAction.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const { emit } = require('../utils');
+
+const waitForAction = (socket, event, requestAction, receiveAction) => (
```
Comment (sklirg) [inline server/utils/socketAction.js]: The websocket emits _events_ on a _channel_, so I'd say the function should be named `waitForEvent` and have `channel` as one of its parameters (instead of event as it has now).

Considering that we use "action" as the name for everything that goes through redux, I'm okay with leaving it as is.

This would also do something with the weird naming like `socket.createAction.action`, which probably should be `socket.createEvent.action` (or `auth` or `admin`), if I'm not mistaken.

TL;DR: naming things is hard

File: server/utils/socketAction.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  new Promise((resolve) => {
+    socket.once(event, async (payload) => {
```
Comment (sklirg) [inline server/utils/socketAction.js]: I'm a bit uncertain on how Node handles these promises if e.g. 1000000 users open and quickly closes a WS connection (as a DoS attempt). NodeJS runtime will GC this if it's not referenced and/or in scope, so it most likely won't be an issue. It could be something to consider researching further, though.

File: server/utils/socketAction.js (hunk 1 above)
Comment (duvholt) [reply server/utils/socketAction.js]: In this case it is literally waiting for an action though. Yes, it uses events to do that, but it only cares about a single action from that event listener. 
And regarding `channel`: both Nodejs and socketio use `eventName` to refer to what I have called `event`. I will consider changing it to eventName though.

I agree about socket.createAction. That started as an ugly hack that grew uglier. I think I'll rename it to emitEvent. Preferably it would just be named emit, but we have to mock that function for testing.

File: server/utils/socketAction.js (hunk 2 above)
Comment (duvholt) [reply server/utils/socketAction.js]: I think it will be garbage collected like everything else in connection when socket is disconnected.
